DSharpPlus/Entities/User/DiscordUser.cs

```diff
@@ -111,7 +126,7 @@ public string AvatarUrl
         /// </summary>
         [JsonIgnore]
         public string DefaultAvatarUrl
-            => $"https://cdn.discordapp.com/embed/avatars/{(this.DiscriminatorInt % 5).ToString(CultureInfo.InvariantCulture)}.png?size=1024";
+            => $"https://cdn.discordapp.com/embed/avatars/{((this.DiscriminatorInt is 0 ? (int)this.Id >> 22 : this.DiscriminatorInt) % 5).ToString(CultureInfo.InvariantCulture)}.png?size=1024";
```

Comment:
For pomelo'd users, this should be % 6: https://discord.com/developers/docs/reference#image-formatting

Also not sure if related because I'm not smart enough, but I seem to be getting invalid results in some cases, e.g. user 1055903982154948680 is giving https://cdn.discordapp.com/embed/avatars/-3.png

Reply:
Wait, dur. I'm casting a 64 bit number to a 32 bit number. Apologies.

Reply:
Try now :sillynod:

Reply:
Theres nothing new to try? :D

Reply:
Did I...push to the wrong upstream?? 
![image](https://github.com/DSharpPlus/DSharpPlus/assets/42438262/42e1a7ba-ac2a-4923-aabb-39f87ce62d01)

Reply:
Drats. I have. One moment 
![image](https://github.com/DSharpPlus/DSharpPlus/assets/42438262/c2061025-2978-4a32-92a2-0f408679de57)

Reply:
You changed it to % 6 for everything, but it only needs to be % 6 for pomelo'd users. If there's a discrim it should still be % 5.
(thanks discord)

Reply:
The only non-pomelo'd users at this point will be bots, but I can change it regardless (also totally didn't forget to reply to this nope)

Reply:
Some people I noticed have also voluntarily not taken up a new name and will await when discord forces them :D but yes it should just be accurate to how Discord treats them, the discrim-specific code can be removed once Discord works out what to do for existing users and bots